study1/week6/longest-substring-without-repeating-characters.java

```diff
@@ -0,0 +1,25 @@
+// https://leetcode.com/problems/longest-substring-without-repeating-characters/description/
+class Solution {
+    public int lengthOfLongestSubstring(String s) {
+        //time complexity is O(n), where n is the lenfth of the String. 
+        //space complexity is O(n), where n is the size of the HashSet.
+
+        int index = 0;
+        int maxLength = 0;
+        Set<Character> set = new HashSet<>();
+
+        while (index < s.length()) {
+            char c = s.charAt(index);
+            if (set.contains(c)) {
+                maxLength = Math.max(maxLength, set.size());
```

Comment:
`maxLength` only changes when index has changes, so move this line after line 19, then you don't need to do `return Math.max(maxLength, set.size());` on line 23, and instead do `return maxLength;`